metron-platform/metron-common/src/main/java/org/apache/metron/common/writer/BulkWriterComponent.java

```diff
@@ -34,21 +34,37 @@
 import java.util.function.Function;
 
 public class BulkWriterComponent<MESSAGE_T> {
-  public static final Logger LOG = LoggerFactory
-            .getLogger(BulkWriterComponent.class);
+  public static final Logger LOG = LoggerFactory.getLogger(BulkWriterComponent.class);
   private Map<String, Collection<Tuple>> sensorTupleMap = new HashMap<>();
   private Map<String, List<MESSAGE_T>> sensorMessageMap = new HashMap<>();
   private OutputCollector collector;
   private boolean handleCommit = true;
   private boolean handleError = true;
+  private Long lastFlushTime;
+  private Long flushIntervalInMs;
+  private boolean flush = false;
+
+
   public BulkWriterComponent(OutputCollector collector) {
     this.collector = collector;
+    this.lastFlushTime = System.currentTimeMillis();
+    this.flush = false;
   }
 
   public BulkWriterComponent(OutputCollector collector, boolean handleCommit, boolean handleError) {
     this(collector);
     this.handleCommit = handleCommit;
     this.handleError = handleError;
+    this.lastFlushTime = System.currentTimeMillis();
```

Comment:
These guys aren't necessary because you're calling the constructor on line 55 which sets them.